Sources/MongoSwift/APM.swift

```diff
@@ -0,0 +1,206 @@
+import Foundation
+import libmongoc
+
+/// A struct representing a server connection, consisting of a host and port.
+public struct ConnectionId {
+    let host: String
+    let port: UInt16
+
+    internal init(_ hostList: UnsafePointer<mongoc_host_list_t>) {
+        var hostData = hostList.pointee
+        self.host = withUnsafeBytes(of: &hostData.host) { (rawPtr) -> String in
+            let ptr = rawPtr.baseAddress!.assumingMemoryBound(to: CChar.self)
+            return String(cString: ptr)
+        }
+        self.port = hostData.port
+    }
+}
+
+/// An event published when a command starts. The event is stored under the key `event`
+/// in the `userInfo` property of `Notification`s posted under the name .commandStarted.
+public struct CommandStartedEvent {
+    /// An internal opaque pointer to a mongoc_apm_command_started_t 
+    internal let _event: OpaquePointer
+
+    /// An internal initializer for a CommandStartedEvent
+    internal init(_ event: OpaquePointer) {
```

Comment:
Rather than storing the opaque pointer here, I think you should "fill out" the struct with the data retrieved from the pointer. It might be unlikely that someone else deletes the pointer, but it _might_ happen.

Also what are the ownership rules for this pointer? Are we responsible for deleting it when we're done with it? If so, then we are presently leaking memory.